### DISCUSSION TITLE
Executing procedures to retrieve data

### DISCUSSION THREAD

**Discussion by GlobexHankScorpio** [vb-consulting/Norm.net#21] [General]
Hello Vedran, I am starting to use Norm since yesterday, I am really courious to play around with it :)

The thing is, let's say I have this simple query, that works really fine:

**var list = connection.Read<Guid, Guid, string, string>("select col1, col2, col3, col4 from incidents").ToList();**

But I would like to move that query to a Procedure and just execute it, like this:

**Norm norm = connection.AsProcedure().Execute("GetAllIncidents");**

Once executed, the Norm object does not have any list result or similar.  Is it possible to retrieve data while executing procedures? For me would be really usefull since I prefer to execute procedures, when possible.

Thanks in advance!

**Comment by vbilopav** [comment]
You should be able to do that without any problems like this:

```csharp
var list = connection
    .AsProcedure()
    .Read<Guid, string, string>("GetAllIncidents")
    .ToList();
```

I'm very late with documentation and a website since I'm very occupied with a paying job, but you can see `AsProcedure` reference here (although still unfinished):
https://norm-dot-net.netlify.app/docs/reference/extension-methods/#asprocedure

**Reply by GlobexHankScorpio** [reply]
Many thanks for your quick answer! I appriciate a lot the work you are doing and sharing it with the community.

Your suggestion works perfectly and fast, love it.

![image](https://github.com/vb-consulting/Norm.net/assets/89591394/6c36f159-b156-4de5-a95a-bff2ac3b5180)

**Comment by vbilopav** [comment]
No problem, I'll leave this discussion open since it's not an issue.
